sdk/trace/src/test/java/io/opentelemetry/sdk/trace/internal/ExtendedSpanProcessorUsageTest.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.trace.internal;
+
+import static io.opentelemetry.sdk.testing.assertj.OpenTelemetryAssertions.assertThat;
+
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.sdk.testing.exporter.InMemorySpanExporter;
+import io.opentelemetry.sdk.trace.ReadWriteSpan;
+import io.opentelemetry.sdk.trace.ReadableSpan;
+import io.opentelemetry.sdk.trace.SdkTracerProvider;
+import io.opentelemetry.sdk.trace.export.SimpleSpanProcessor;
+import org.junit.jupiter.api.Test;
+
+/** Demonstrating usage of {@link ExtendedSpanProcessor}. */
+public class ExtendedSpanProcessorUsageTest {
+
+  private static final AttributeKey<String> FOO_KEY = AttributeKey.stringKey("foo");
+  private static final AttributeKey<String> BAR_KEY = AttributeKey.stringKey("bar");
+
+  private static class CopyFooToBarProcessor implements ExtendedSpanProcessor {
+
+    @Override
+    public void onStart(Context parentContext, ReadWriteSpan span) {}
+
+    @Override
+    public boolean isStartRequired() {
+      return false;
+    }
+
+    @Override
+    public void onEnd(ReadableSpan span) {}
+
+    @Override
+    public boolean isEndRequired() {
+      return false;
+    }
+
+    @Override
+    public void onEnding(ReadWriteSpan span) {
+      String val = span.getAttribute(FOO_KEY);
+      span.setAttribute(BAR_KEY, val);
+    }
+
+    @Override
+    public boolean isOnEndingRequired() {
+      return true;
+    }
+  }
+
+  @Test
+  public void testExtendedSpanProcessorUsage() {
```

Comment:
#nit: the "test" prefix is redundant since the method is annotated with `@Test`
```suggestion
  public void extendedSpanProcessorUsage() {
```